src/gui/mainwindow.cpp

```diff
@@ -1897,35 +1897,54 @@ void MainWindow::on_actionDownloadFromURL_triggered()
 }
 
 #if defined(Q_OS_WIN) || defined(Q_OS_MACOS)
-void MainWindow::handleUpdateCheckFinished(bool updateAvailable, QString newVersion, bool invokedByUser)
+void MainWindow::handleUpdateCheckFinished(const bool updateAvailable, const QString &newVersion, const bool invokedByUser)
 {
-    QMessageBox::StandardButton answer = QMessageBox::Yes;
-    if (updateAvailable)
+    m_ui->actionCheckForUpdates->setEnabled(true);
+    m_ui->actionCheckForUpdates->setText(tr("&Check for Updates"));
+    m_ui->actionCheckForUpdates->setToolTip(tr("Check for program updates"));
+
+    if (!updateAvailable && invokedByUser)
     {
-        answer = QMessageBox::question(this, tr("qBittorrent Update Available")
-            , tr("A new version is available.") + "<br/>"
-                + tr("Do you want to download %1?").arg(newVersion) + "<br/><br/>"
-                + QString::fromLatin1("<a href=\"https://www.qbittorrent.org/news.php\">%1</a>").arg(tr("Open changelog..."))
-            , QMessageBox::Yes | QMessageBox::No, QMessageBox::Yes);
-        if (answer == QMessageBox::Yes)
+        auto *msgBox = new QMessageBox {QMessageBox::Information, QLatin1String("qBittorrent")
+            , tr("No updates available.\nYou are already using the latest version.")
+            , QMessageBox::Ok, this};
+        msgBox->setAttribute(Qt::WA_DeleteOnClose);
+        connect(msgBox, &QDialog::finished, this, [this](const int)
         {
-            // The user want to update, let's download the update
-            ProgramUpdater *updater = dynamic_cast<ProgramUpdater * >(sender());
-            updater->updateProgram();
-        }
+            if (Preferences::instance()->isUpdateCheckEnabled())
+                m_programUpdateTimer->start();
+        });
+        msgBox->open();
+        return;
     }
-    else if (invokedByUser)
+
+    if (updateAvailable)
```

Comment:
I would reformat it to the following form:
```c++
if (updateAvailable)
{
}
else if (invokedByUser)
{
}
```

Reply:
Ok, reverted.